frontend/app/page.jsx

```diff
@@ -0,0 +1,32 @@
+'use client';
+import { useEffect } from "react";
+import { useRouter } from 'next/navigation';
+import { auth, initFirebaseAuth } from '../firebase/config';
+import { onAuthStateChanged } from 'firebase/auth';
+import Cookies from "js-cookie";
+import Login from './login/page';
+import styles from '../styles/Home.module.css';
+
+const App = () => {
+    initFirebaseAuth();
+    const router = useRouter();
+
+    useEffect(() => {
```

Comment:
Good opportunity for a custom hook! This is not necessary at all, but if you factor this function out into its own file, you can create a custom hook and call it in this file. That will make your homepage code a little more clean, and more scaleable down the road.